src/common-utils/scripts/bash_theme_snippet.sh

```diff
@@ -23,3 +22,20 @@ __bash_prompt() {
 }
 __bash_prompt
 export PROMPT_DIRTRIM=4
+
+# Function to set the terminal title to the command being executed
+preexec() {
+    local cmd=$(history 1 | sed 's/^[ ]*[0-9]*[ ]*//')
+    echo -ne "\033]0;${USER}@${HOSTNAME}: ${cmd}\007"
+}
+
+# Function to reset the terminal title to the shell type after the command is executed
+precmd() {
+    echo -ne "\033]0;${USER}@${HOSTNAME}: ${SHELL}\007"
+}
+
+# Trap DEBUG signal to call preexec before each command
+trap 'preexec' DEBUG
+
+# Set the PROMPT_COMMAND to call precmd before displaying the prompt
+PROMPT_COMMAND='precmd'
```

Comment:
Can we append to the PROMPT_COMMAND instead of overwriting? because if the user has other commands set in PROMPT_COMMAND, they will be lost.

Reply:
Appending it now